ui/elements/IconGroup/IconGroup.tsx

```diff
@@ -0,0 +1,53 @@
+import * as React from 'react';
+import classnames from 'classnames';
+import { StyledIconGroup } from './IconGroup.styles';
+import { IconGroupProps } from './types';
+import Icon from '../Icon/Icon';
+
+/**
+ * IconGroup is a component that groups multiple Icon components together.
+ */
+const IconGroup: React.FC<IconGroupProps> = ({
+    icons,
+    children,
+    className,
+    size,
+    color = 'grey',
+    horizontal = true,
+    spacing = '8px',
+    ...props
+}) => {
+    const classes = classnames('grauity-icon-group', className);
+
+    // Render icons passed as array or children
+    const renderIcons = () => {
+        if (icons && icons.length > 0) {
+            return icons.map((iconProps: IconGroupProps['icons'][0], index: number) => {
```

Comment:
can directly use IconProps here (add required import too pls)
```suggestion
            return icons.map((iconProps: IconProps, index: number) => {
```